JunList.java

```diff
@@ -19,99 +21,104 @@ public void append(int value) {
             int[] newList = new int[list.length * 2];
             for (int i = 0; i < list.length; i++) {
                 newList[i] = this.list[i];
-                this.list = newList;
             }
-            this.list[trackListPosition] = value;
-            trackListPosition += 1;
+            this.list = newList;
         }
+        this.list[trackListPosition] = value;
+        trackListPosition += 1;
+    }
 
-        public void add(int position,int value){
-            //creates list twice as big when necessary
-            if (trackListPosition == list.length) {
-                int[] newList = new int[list.length * 2];
-                for (int i = 0; i < list.length; ++i) {
-                    if (i < position) {
-                        newList[i] = list[i];
-                    } else if (position == i) {
-                        newList[i] = value;
-                    } else {
-                        newList[i] = list[i - 1];
-                    }
-                }
-                this.list = newList;
-            } else {
-                for (int i = trackListPosition; i > position; i--) {
-                    list[i] = list[i - 1];
+    public int get(int index) {
+        return this.list[index];

```

Comment:
this is unneccesary, what happens if someone inputs 10000 and you only have 5 elements?

Reply:
added out of bound exception if statement